pkg/build/controller/config_controller.go

```diff
@@ -33,11 +39,66 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
+
+	JenkinsConfig serverapi.JenkinsPipelineConfig
+
+	// recorder is used to record events.
+	Recorder record.EventRecorder
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		svcName := c.JenkinsConfig.ServiceName
+		if len(svcName) == 0 {
+			return fmt.Errorf("the Jenkins Pipeline ServiceName must be set in master configuration")
+		}
+
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(svcName); err == nil {
```

Comment:
only triggering off the service means that other parts of the template (backing pod/rc, etc) could be deleted, or the service could be messed up, and the system won't recover